packages/shared/src/components/plus/PlusIOS.tsx

```diff
@@ -55,15 +56,27 @@ export const PlusIOS = ({
       displayToast(DEFAULT_ERROR);
     });
 
+    promisifyEventListener<void, 'true' | 'false'>(
+      'iap-loading',
+      ({ detail }) => {
+        setIsLoading(detail.toLowerCase() === 'true');
```

Comment:
Mostly worried for this to get stuck loading forever

Reply:
on promise reject `setIsLoading(false)`?

Reply:
I mean on error it will throw, which is rejection

Reply:
Wrapped it in try-catch, does that help?